exercises/pig-latin/src/test/java/PigLatinTranslatorTest.java

```diff
@@ -1,85 +1,95 @@
+import org.junit.Before;
+import org.junit.Ignore;
 import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-
-import java.util.Arrays;
-import java.util.Collection;
 
 import static org.junit.Assert.assertEquals;
 
-@RunWith(Parameterized.class)
 public class PigLatinTranslatorTest {
 
-    private String englishPhrase;
-    private String pigLatinTranslation;
+    private PigLatinTranslator pigLatinTranslator;
 
-    @Parameterized.Parameters(name = "{index}: expected \"{0}\" to translate to the pig latin phrase \"{1}\"")
-    public static Collection<Object[]> data() {
-        return Arrays.asList(new Object[][]{
-                // Ay is added to words that start with vowels
-                {"apple", "appleay"},
-                {"ear", "earay"},
-                {"igloo", "iglooay"},
-                {"object", "objectay"},
-                {"under", "underay"},
+    public PigLatinTranslatorTest() {
+    }
 
-                // Word beginning with vowel follwed by a 'qu'
-                {"equal", "equalay"},
+    @Before
+    public void setup() {
+        pigLatinTranslator = new PigLatinTranslator();
+    }
 
-                // First letter and ay are moved to the end of words that start with consonants
-                {"pig", "igpay"},
-                {"koala", "oalakay"},
-                {"xenon", "enonxay"},
+    @Test
+    public void testWordsStartWithVowels() {
+        assertEquals("appleay", pigLatinTranslator.translate("apple"));
```

Comment:
It may make more sense to separate each of these `assertEquals` statements out into individual tests. @exercism/java what do you guys think? 🙂

Reply:
Actually I believe they should be group together since they all belong to the same type?

Reply:
It depends how we want to think about this - I personally prefer tests to be as granular as possible. For exercises like this, where simply checking if words start with vowels or consonants isn't actually sufficient to 'truly' convert to pig latin (there isn't one easy rule), I think it can be of benefit to guide the user a little more carefully.

For example, the assertions in `testWordsWithPrefixTreatedAsSingleConsonant()` are actually an amalgamation of a handful of exceptions to the 'easy' rules.

I'd be interested in hearing everyone else's views on this - I have a particular vendetta against this exercise, so maybe I am just being picky :wink:

Reply:
I agree with @sjwarner-bp, I think each test should only test one thing. This is for several reasons:
1. Since we're promoting TDD, if each test only tests one thing (in this case one word) it'll be much easier to keep track of your progress by how many tests pass.

2. I think it's quite rewarding when you make a test pass and you wouldn't  get that as often if each test has several assertions.

3. We should match the [canonical data](https://github.com/exercism/problem-specifications/blob/master/exercises/pig-latin/canonical-data.json) as closely as possible, and the canonical data only tests one word at a time. Furthermore, the people making the canonical data try to make sure that each word tests a different thing and that they're in a good order. That is, if you test each word individually, it shouldn't be the case that if the test for one word passes then the tests for some subsequent words will definitely pass. So the canonical data is deliberately constructed so that each word should have its own test.

Does that make sense? 🙂

Reply:
Hi, thanks for the clarification! Yes these are really reasonable. I will make the changes accordingly for each PR.